### PR TITLE
Network: fix goroutine leak on shutdown in v2 protocol

### PR DESCRIPTION
**Pull request by reinkrul**
Part of https://github.com/nuts-foundation/nuts-node/issues/1900

### CODE REVIEW

File: network/transport/v2/protocol.go [hunk 1]
```diff
@@ -187,13 +189,28 @@ func (p *protocol) Configure(_ transport.PeerID) error {
...
+		p.routines.Add(1)
+		go func(w *sync.WaitGroup) {
+			defer w.Done()
+			p.diagnosticsMan.start(p.ctx, time.Duration(p.config.DiagnosticsInterval)*time.Millisecond)
+		}(p.routines)
 	}
 
-	p.listHandler = newTransactionListHandler(p.ctx, p.handleTransactionList)
-	p.listHandler.start()
+	ctx := p.ctx // use copy of pointer to avoid data race
```
Comment (woutslakhorst) [inline network/transport/v2/protocol.go:202]: copy of pointer still points to same instance.

File: network/transport/v2/protocol.go [hunk 2]
```diff
@@ -187,13 +189,28 @@ func (p *protocol) Configure(_ transport.PeerID) error {
...
+	// Wrap listHandler function to supply the context to handleTransactionList.
+	// It would be prettier to pass a context around in all protocol message handlers (since most use context.Background() for database access now),
+	// but that is too big a change for now.
+	p.listHandler = newTransactionListHandler(p.ctx, func(peer transport.Peer, envelope *Envelope) error {
```
Comment (woutslakhorst) [inline network/transport/v2/protocol.go:206]: change the `handleFunc` signature so the ctx is passed as well and you don't need a copy. HandleTransactionList should reuse the context passed by the TransctionListHandler, not the context from the protocol.

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
👍 on using contexts to stop procedures

File: network/transport/v2/peer_diagnostics_test.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ func Test_PeerDiagnosticsManager(t *testing.T) {
 
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
-		manager.start(ctx, 5*time.Millisecond)
+		go manager.start(ctx, 5*time.Millisecond)
 
 		// Wait for the calls
 		wg.Done()
```
Comment (gerardsn) [inline network/transport/v2/peer_diagnostics_test.go:45]: unrelate to this PR, but this test is broken. it doesn't wait...

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**